tonic/parameters.py

```
@@ -0,0 +1,158 @@
+            These are symbolic objects generated by Geometric Entities.
+            If a displacement field is assosicated with a ChildParameter,
+            the parent Parameter should take care of the dependent disp-
+            lacement field(s).
+        2. Parameter attribute is_variable
+            I can see why we might need it, but perhaps the concept
+            should be more clear.
+"""
+
+import sympy
+
+
+class Parameter:
+    _parameters_counter = 1
+
+    def __init__(self, value, is_variable=True, name="p_"):
+        self.value = value
+        self.is_variable = is_variable
+
+        self.symbol = sympy.var(name + str(Parameter._parameters_counter))
+        Parameter._parameters_counter += 1
+
+        self.children = []
+
+    def __str__(self):
+        _parameter_info = {
+            "name": self.symbol,
+            "value": self.value,
+            "is_variable": self.is_variable,
+        }
+        return str(_parameter_info) + "\n"
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __add__(self, other):
+        """
+        Addition rules:
+            - Parameter + int / float -> ChildParameter
+                Adding number to a parameter does not introduce a new
+                parameter, but creates a connected (child) parameter.
+                When the parent's value changes, the child should also
+                change.
+            - Parameter + Parameter -> float
+                There is no point of creating a new Parameter
+                object when adding one variable parameters to another.
+                The construction is meaningless, but for debugging purpose
+                it is possible.
+        """
+        if isinstance(other, int) or isinstance(other, float):
+            return ChildParameter(self, self.symbol + other)
+        elif isinstance(other, Parameter) or isinstance(other, ChildParameter):
+            return self.value + other.value
```

This should throw an error in "production" code I think, so maybe raise a warning currently?

---

Two thoughts:
1. I agree adding two variable parameters is meaningless. Adding a variable parameter to a fixed parameter does have meaning though. I guess it should return a `ChildParameter` of the variable parameter. There needs to be some logic to do this. (shout if I'm wrong here)

2. I'm not a massive fan of `__add__` etc. returning fundamentally different things. I feel we should always return something that inherits from the `Parameter` even if this is just for debugging.